### ISSUE TITLE
onSlideChangeEnd does not trigger between very fast swipes

### ISSUE THREAD

**Issue by easwee** [nolimits4web/Swiper#1403]
If you swipe 2x or more times fast enough, that slides won't stop the transition animation between swipes, only onSlideChangeStart will trigger 2x, but onSlideChangeEnd will trigger only once - at the end of swipes when animation stops.

Looking in swiper source this is probably due to the fact that transitionEnd is responsible for triggering onSlideChangeEnd, but if the transition doesn't end between slides, onSlideChangeEnd will never trigger.

Is it possible to capture onSlideChangeEnd in this case? Is there another event I could listen to, that would trigger on each slide sliding end. (onTransitionEnd is not an options since it will trigger even if the active slide won't change and you only moved a slide a bit and released without triggering slide change).

Simple test sample (just swipe very fast): 

            swiper = new Swiper('#swiper', {
                onSlideChangeStart: function() {
                    console.log('Slide change START');
                },
                onSlideChangeEnd: function() {
                    console.log('Slide change END');
                }
            });

**Comment by wgroenewold** [comment]
To me this sounds like the same issue as nolimits4web/Swiper#1401 and nolimits4web/Swiper#1402. Nevertheless good to address another point where it lacks and needs some improvement.

**Comment by easwee** [comment]
Yes - it's very similar. The problematic part in swiper source is:

        s.onTransitionEnd = function (runCallbacks) {
            s.animating = false;
            s.setWrapperTransition(0);
            if (typeof runCallbacks === 'undefined') runCallbacks = true;
            if (s.lazy) s.lazy.onTransitionEnd();
            if (runCallbacks) {
                s.emit('onTransitionEnd', s);
                if (s.activeIndex !== s.previousIndex) {
                    s.emit('onSlideChangeEnd', s);
                }
            }
            if (s.params.hashnav && s.hashnav) {
                s.hashnav.setHash();
            }        
        };

As you can see onSlideChangeEnd has no chance to trigger until the transition is going on - 2 fast swipes will keep one unterminated transition effect and won't capture the slide change.

**Comment by nolimits4web** [comment]
Closing as duplicate of #1216. There will no be onSlideChangeEnd in this case as there will no be transitionend event triggered. Just change your logic to onSlideChangeStart

**Comment by andyiwest** [comment]
But

onSlideChangeStart: function() {
    console.log('slide changing');
},

doesn't even work?
